### PR TITLE
Use prow to push container at release

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
To reduce the number of system we use let's move the release process
totally to prow.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/release.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash -e
 
+make IMAGE_TAG=$TAG docker-build docker-push
```
Comment (RamLavi) [inline hack/release.sh:3]: isn't this redundant ? we also do it in the travis.yml

Comment (qinqon) [reply hack/release.sh:3]: This is for release what we have at .travis.yaml after this PR is at mater

Comment (RamLavi) [reply hack/release.sh:3]: yeah sorry total brain-fart

**Comment by qinqon** [comment]
/override pull-e2e-cluster-network-addons-operator-kubemacpool-functests
/override pull-e2e-cluster-network-addons-operator-lifecycle-k8s
/override pull-e2e-cluster-network-addons-operator-nmstate-functests
/override pull-e2e-cluster-network-addons-operator-workflow-k8s

**Pull request merged**